### PR TITLE
Force use of Python 3 for all scripts.

### LINKED ISSUE #251: Python -> Python3
It would be nice to standardise on Python3, currently a mixture of python and python3, used for various x86 and stm32 ctests.

### PR DESCRIPTION
**Pull request by tmiw**
Resolves #251 by substituting calls to "python" or "python2" with "python3".

### CODE REVIEW

**Comment by drowe67** [comment]
Thanks @tmiw that's fast work!

File: stm32/support/hex2dfu.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
 #
 # Intel HEX to ST microelectronics DfuSe file converter
 # Copyright (C)2015 Thomas Kindler <mail@t-kindler.de>
```
Comment (drowe67) [inline stm32/support/hex2dfu.py:4]: @srsampson I was wondering about the codec2/stm/support Python scripts.  Git log suggested they came from you.  I've never used them and wondered what they are for? Is there any way to test the change to python3 in this PR?

**Comment by drowe67** [comment]
I just tried this PR on bear with the stm32 ctests and all good :+1:

File: stm32/support/hex2dfu.py (hunk 1 above)
Comment (drowe67) [reply stm32/support/hex2dfu.py:4]: Ahh, OK we do use those utilities, and looks like we broke something:
```
cd codec2/stm32/build_stm32/
make clean && make
<snip>
[ 63%] Linking C executable sm1000v5.elf
Creating binary for sm1000v5
Creating dfu file for sm1000v5
  File "/home/david/codec2/stm32/support/hex2dfu.py", line 32
    print "Loading %s..." % args.source
          ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Loading %s..." % args.source)?
```

Comment (tmiw) [reply stm32/support/hex2dfu.py:4]: That particular issue should be resolved now.

**Comment by drowe67** [comment]
On https://github.com/drowe67/codec2/pull/254/commits/ec7b3e28e1885bed41f5e938a1b3fe3c632f7690 I'm still seeing:
```
david@bear:~/codec2/stm32/build_stm32 ms-python3 $ make
[  1%] Built target codec2_native
[  2%] Built target stm32f4_adac
[ 46%] Built target CMSIS
[ 52%] Built target stm32f4
[ 52%] Linking C executable sm1000v5.elf
Creating binary for sm1000v5
Creating dfu file for sm1000v5
  File "/home/david/codec2/stm32/support/hex2dfu.py", line 42
    except Exception, e:
                    ^
SyntaxError: invalid syntax
```
Not sure if I'm doing something wrong. "bear" is on if you'd like to try it.  Might be machine or OS specific?  I'm not across exception handling in Python.

**Comment by drowe67** [comment]
Ooops sorry, I've made sure bear is on.  I'm seeing a similar issue on https://github.com/drowe67/codec2/pull/254/commits/a223e5f09d1fb271de98064dbfd063e1b64f72c9 :thinking:

**Comment by tmiw** [comment]
All STM32 ctests should be good now. 👍

**Comment by drowe67** [comment]
Thanks @tmiw.  Given there has been a few changes to the `hex2dfu` utility, it would be good to try the output `sm1000v5.dfu` file, which is used by the Windows centric flashing tools.  Perhaps we could ask Walter to try it out on his SM1000?

**Comment by tmiw** [comment]
Actually, should we stop generating .dfu files altogether? DfuSeUtil seems to be deprecated by ST in favor of STM32CubeProgrammer (shown below) and the latter seems to use .bin files just like dfu-util.

Screen Shot 2021-11-11 at 11 00 48 AM

(I tried to get DfuSeUtil running on my Windows VM but it wouldn't detect the SM1000 even though Windows did. If we still want to support DfuSeUtil, we can send out the generated .dfu file anyway for test.)

**Comment by drowe67** [comment]
That's a fine idea @tmiw, so I think we'd need to:
1. Remove the `.dfu` Makefile targets.
2. Modify the SM1000 Manual with instructions to use STM32CubeProgrammer.
3. rm the codec2/stm32/support files if not needed.
4. Work with people in the stm32 community to test.

However that feels like a significant chunk of work to me, orthogonal to this PR - which is around python3. It will involve some interaction with the community to make sure Windows users are comfortable reflashing `.bin` files.

How about we merge this PR, then consider the `.dfu` removal above in a future PR?

**Comment by tmiw** [comment]
Another PR is fine for the DFU removal. 👍

**Comment by tmiw** [comment]
PR for the DFU removal created.

**Pull request merged**